rfcs/text/0016_buildkite.md

```diff
@@ -0,0 +1,907 @@
+| Stable                               | No      | Yes       | No             | Yes      |
+| Surfaces information intuitively     | No      | Yes       | No             | Yes      |
+| Pipelines                            | Yes     | Yes       | Yes            | Yes      |
+| Advanced Pipeline logic              | Yes     | Yes       | Partial        | Partial  |
+| Cloud-friendly pricing model         | Yes     | Yes       | Yes            | No       |
+| Public access                        | Yes     | Yes       | Yes            | Partial  |
+| Secrets handling                     | Yes     | No        | Yes            | Partial  |
+| Support or Documentation             | No      | Yes       | Yes            | Partial  |
+| Scheduled Builds                     | Yes     | Yes       | Yes            | Yes      |
+| Container support                    | Partial | Yes       | Yes            | Yes      |
+|                                      |         |           |                |          |
+| Customization                        | No      | Yes       | No             | No       |
+| Core functionality is first-party    | No      | Mostly    | Mostly         | TODO     |
+| First-class support for test results | Buggy   | No        | No             | Yes      |
+| GitHub Integration                   | Yes     | Limited   | Yes            | Yes      |
+| Local testing / reproduction?        | TODO    | TODO      | TODO           | TODO     |
+
+TODO link the conclusions to each section?
+
+# Motivation
+
+We have lived with the scalability and stability problems of our current Jenkins infrastructure for several years. We have spent a significant amount of time designing around problems, and are limited in how we can design our pipelines. Since the company-wide effort to move to a new system has been cancelled for the forseeable future, we are faced with either re-engineering the way we use Jenkins, or exploring other solutions and potentially managing one ourselves.
+
+This RFC is focused on the option of using a system other than Jenkins, and managing it ourselves (to the extent that it must be managed). If the RFC is rejected, the alternative will be to instead invest significantly into Jenkins to further stabilize and scale our usage of it.
+
+## Required and Desired Capabilities
+
+### Required
+
+#### Scalable
+
+- Able to run 100s of pipelines and 1000s of individual steps in parallel without issues.
+- If scaling agents/hosts is self-managed, dynamically scaling up and down based on usage should be supported and reasonably easy to do.
+
+#### Stable
+
+- Every minute of downtime can affect 100s of developers.
+- For systems provided as a service, they should not have frequent outages. This is a bit hard to define. 1-2 hours of downtime, twice a month, during peak working hours, is extremely disruptive. 10 minutes of downtime once or twice a week can also be very disruptive, as builds might need to be re-triggered, etc.
```

Comment:
Not sure if this is the right place, but it might be worth noting that our team can not be on-call, so our responsibility in regards to stability needs to be limited.